training/src/anemoi/training/commands/mlflow.py

```
@@ -129,7 +173,51 @@ def run(args: argparse.Namespace) -> None:
                 args.export_deleted_runs,
                 log_level,
             ).sync()
-            return
+            return None
+
+        if args.subcommand == "prepare":
+            # Load configuration
+            with initialize(version_base=None, config_path="./"):
+                cfg = compose(config_name=args.config_name)
+
+            # Create MLflow client and get experiment
+            client = AnemoiMlflowClient(cfg.diagnostics.log.mlflow.tracking_uri, authentication=True)
+            experiment_id = client.get_experiment_by_name(cfg.diagnostics.log.mlflow.experiment_name).experiment_id
+
+            # Parse configuration
+            if cfg.training.run_id is not None:  # Existing run_id
+                LOGGER.info("Existing run_id: %s", cfg.training.run_id)
+                try:
+                    client.get_run(cfg.training.run_id)
+                except ValueError as e:
+                    e(f"Run_id {cfg.training.run_id} not found. Modify your configuration file.")
+                    return None
```

Here you should raise an exception, not return. Something like this:

```python
raise ValueError("run id not found...") from e
```